### PR TITLE
[Codemod: new] replace useSearchParams

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: next/13/replace-get-search-params/config.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+	"schemaVersion": "1.0.0",
+	"name": "next/13/replace-get-search-params",
```
Comment (grzpab) [inline next/13/replace-get-search-params/config.json:3]: replace use search params

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,6220 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (grzpab) [inline yarn.lock:1]: you used yarn instead of pnpm I think :D

File: next/13/replace-get-search-params/index.ts [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+type ParsedUrlQuery = Record<string, string | string[]>;
+
+// ref: https://github.com/vercel/next.js/blob/ab7b0f59fb0d793fb4ee593ae93f7e08bab23b5b/packages/next/src/shared/lib/router/utils/querystring.ts#L3C17-L3C39
+
+export const searchParamsToUrlQuery = (
+    searchParams: ReadonlyURLSearchParams
+  ): ParsedUrlQuery => {
+    const query: ParsedUrlQuery = {}
+
+    searchParams.forEach((value, key) => {
```
Comment (grzpab) [inline next/13/replace-get-search-params/index.ts:51]: I think we need to add the licensing information from Vercel.
on the other hand, can we import `searchParamsToUrlQuery`?

File: next/13/replace-get-search-params/index.ts [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  
+export const useCompatSearchParams = () => {
+    const _searchParams = useSearchParams();
+    const params = useParams() ?? {};
+
+    const searchParams =  searchParamsToUrlQuery(_searchParams);
+
+    const paramsAsObj = {
+      ...searchParams, 
+      // Merge params into \`query\`, overwriting any specified in search
```
Comment (grzpab) [inline next/13/replace-get-search-params/index.ts:72]: you can remove this comment

File: next/13/replace-get-search-params/index.ts [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+        query[key] = [query[key] as string, value]
+      }
+    })
+    return query
+  }
+
+  
+export const useCompatSearchParams = () => {
+    const _searchParams = useSearchParams();
+    const params = useParams() ?? {};
```
Comment (grzpab) [inline next/13/replace-get-search-params/index.ts:66]: this will be null if used in pages

File: next/13/replace-get-search-params/index.ts (hunk 3 above)
Comment (DmytroHryshyn) [reply next/13/replace-get-search-params/index.ts:51]: no they dont export it.

**Review by grzpab — APPROVED** [review]

**Pull request merged**